ADAL/src/telemetry/ADTelemetryHttpEvent.m

```diff
@@ -22,42 +22,80 @@
 // THE SOFTWARE.
 
 #import "ADTelemetryHttpEvent.h"
+#import "ADTelemetryEventStrings.h"
 
 @implementation ADTelemetryHttpEvent
 
 - (void)setHttpMethod:(NSString*)method
 {
-    [self setProperty:@"http_method" value:method];
+    [self setProperty:AD_TELEMETRY_HTTP_METHOD value:method];
 }
 
 - (void)setHttpPath:(NSString*)path
 {
-    [self setProperty:@"http_path" value:path];
+    [self setProperty:AD_TELEMETRY_HTTP_PATH value:path];
+}
+
+- (void)setHttpRequestIdHeader:(NSString*)requestIdHeader
+{
+    [self setProperty:AD_TELEMETRY_HTTP_REQUEST_ID_HEADER value:requestIdHeader];
 }
 
 - (void)setHttpResponseCode:(NSString*)code
 {
-    [self setProperty:@"http_response_code" value:code];
+    [self setProperty:AD_TELEMETRY_HTTP_RESPONSE_CODE value:code];
+}
+
+- (void)setOAuthErrorCode:(NSString*)code
+{
+    [self setProperty:AD_TELEMETRY_OAUTH_ERROR_CODE value:code];
 }
 
 - (void)setHttpResponseMethod:(NSString*)method
 {
-    [self setProperty:@"http_response_method" value:method];
+    [self setProperty:AD_TELEMETRY_HTTP_RESPONSE_METHOD value:method];
 }
 
 - (void)setHttpRequestQueryParams:(NSString*)params
 {
-    [self setProperty:@"request_query_params" value:params];
+    [self setProperty:AD_TELEMETRY_REQUEST_QUERY_PARAMS value:params];
 }
 
 - (void)setHttpUserAgent:(NSString*)userAgent
 {
-    [self setProperty:@"user_agent" value:userAgent];
+    [self setProperty:AD_TELEMETRY_USER_AGENT value:userAgent];
 }
 
 - (void)setHttpErrorDomain:(NSString*)errorDomain
 {
-    [self setProperty:@"http_error_domain" value:errorDomain];
+    [self setProperty:AD_TELEMETRY_HTTP_ERROR_DOMAIN value:errorDomain];
+}
+
+- (void)processEvent:(NSMutableDictionary*)eventToBeDispatched
+{
+    [super processEvent:eventToBeDispatched];
+    
+    (void)eventToBeDispatched;
+    
+    int httpEventCount = 1;
+    if ([eventToBeDispatched objectForKey:AD_TELEMETRY_HTTP_EVENT_COUNT])
+    {
+        httpEventCount = [[eventToBeDispatched objectForKey:AD_TELEMETRY_HTTP_EVENT_COUNT] intValue] + 1;
+    }
+    [eventToBeDispatched setObject:[NSString stringWithFormat:@"%d", httpEventCount] forKey:AD_TELEMETRY_HTTP_EVENT_COUNT];
+    
+    NSArray* properties = [self getProperties];
+    for (NSArray* property in properties)
+    {
+        if ([property[0] isEqualToString:AD_TELEMETRY_HTTP_RESPONSE_CODE]
+            ||[property[0] isEqualToString:AD_TELEMETRY_OAUTH_ERROR_CODE]
+            ||[property[0] isEqualToString:AD_TELEMETRY_HTTP_ERROR_DOMAIN]
+            ||[property[0] isEqualToString:AD_TELEMETRY_HTTP_PATH]
+            ||[property[0] isEqualToString:AD_TELEMETRY_HTTP_REQUEST_ID_HEADER])
+        {
+            [eventToBeDispatched setObject:property[1] forKey:property[0]];
```

Comment:
This isn't really aggregating anything. It's collapsing and overriding data. Is this really what was spec'd?